File: Examples/TS/concat.ts
function size<A>(l: A[]): number {
    if (l.length === 0) {
        return 0;
    } else {
        const t= l.slice(1);
        return (1 + size(t));
    }
}
function concat<A>(l1: A[], l2: A[]): A[] {
    if (l1.length === 0) {
        return l2;
    } else {
        const h= l1[0];
        const t= l1.slice(1);
        return [h, ...concat(t, l2)];
    }
}
const l1 = [5, 4, 3, 2, 1]
const l2 = [6, 7, 8, 9, 0]
const r = concat(l1, l2)

File: Examples/PY/concat.py
def size[A](l: list[A]) -> int:
    match l:
        case [_, *t]:
            return 1 + (size(t))
        case _:
            return 0
def concat[A](l1: list[A], l2: list[A]) -> list[A]:
    match l1:
        case [h, *t]:
            return [h, *(concat(t, l2))]
        case _:
            return l2
l1 = [5, 4, 3, 2, 1]
l2 = [6, 7, 8, 9, 0]
r = concat(l1, l2)
